examples/langchain/retrieval_qa_chain.py

```diff
@@ -0,0 +1,45 @@
+import mlflow
+import os
+import tempfile
+
+from langchain.chains import RetrievalQA
+from langchain.document_loaders import TextLoader
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.llms import OpenAI
+from langchain.text_splitter import CharacterTextSplitter
+from langchain.vectorstores import FAISS
+from mlflow.langchain import _LOADER_FN_KEY, _PERSIST_DIR_KEY
+
+assert "OPENAI_API_KEY" in os.environ, "Please set the OPENAI_API_KEY environment variable."
+persist_dir = tempfile.mkdtemp()
```

Comment:
We should avoid using `mkdtemp` unless we really need it because it doesn't clean up the directory. Instead, we should use `with tempfile.TemporaryDirectory():`.